src/runtimes/wapc.rs

```diff
@@ -229,6 +236,84 @@ fn send_request_and_wait_for_response(
     }
 }
 
+pub(crate) struct WapcStack {
+    engine: wasmtime::Engine,
+    module: wasmtime::Module,
+    epoch_deadlines: Option<crate::policy_evaluator_builder::EpochDeadlines>,
+    wapc_host: wapc::WapcHost,
+}
+
+impl WapcStack {
+    pub(crate) fn new(
+        engine: wasmtime::Engine,
+        module: wasmtime::Module,
+        epoch_deadlines: Option<crate::policy_evaluator_builder::EpochDeadlines>,
+    ) -> Result<Self> {
+        let wapc_host = Self::setup_wapc_host(engine.clone(), module.clone(), epoch_deadlines)?;
+
+        Ok(Self {
+            engine,
+            module,
+            epoch_deadlines,
+            wapc_host,
+        })
+    }
+
+    /// Provision a new wapc_host. Useful for starting from a clean slate
```

Comment:
```suggestion
    /// Provision a new wapc_host. Useful for starting from a clean state
```

Reply:
This is an [English way of saying](https://www.wordnik.com/words/clean%20slate) that means:

* > noun by extension a fresh start
* >  an opportunity to start over without prejudice

Reply:
I can change it to the more formal way you proposed

Reply:
Oh, that's nice! Cool new expression learnt. Thanks! No need for change.